### PR TITLE
Sync in desktop

### PR DESCRIPTION
**Pull request by jessopb**
Brings tags in from redux
Brings sync in from lbryinc
Snackbar when doing preference actions (follow, etc) before wallet preferences populated

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -136,8 +136,8 @@
     "imagesloaded": "^4.1.4",
     "json-loader": "^0.5.4",
     "lbry-format": "https://github.com/lbryio/lbry-format.git",
-    "lbry-redux": "lbryio/lbry-redux#ba5d6b84bec6bdb2f0a1a6b23e695212c65f650e",
-    "lbryinc": "lbryio/lbryinc#db0663fcc4a64cb082b6edc5798fafa67eb4300f",
+    "lbry-redux": "lbryio/lbry-redux#da37314b24e82c8c348f463818a74ac21b4e3f38",
```
Comment (seanyesmunt) [inline package.json:139]: Is there a lbry-redux PR for this?

File: ui/redux/actions/settings.js [hunk 2]
```diff
@@ -85,7 +98,18 @@ export function doClearDaemonSetting(key) {
...
 // if doPopulate is applying settings, we don't want to cause a loop; doNotDispatch = true.
 export function doSetDaemonSetting(key, value, doNotDispatch = false) {
-  return dispatch => {
+  return (dispatch, getState) => {
+    const state = getState();
+    const ready = selectPrefsReady(state);
+
+    if (!ready) {
+      return dispatch(
+        doToast({
```
Comment (seanyesmunt) [inline ui/redux/actions/settings.js:107]: This should be a separate function that can be re-used. 
```
if (!ready) {
  return dispatch(doAlertSyncNotReady())
}
```

(not sure about the name)

File: ui/redux/actions/settings.js [hunk 3]
```diff
@@ -200,6 +236,7 @@ export function doSetWalletSyncPreference(pref) {
 }
 
 export function doPushSettingsToPrefs() {
+  // if syncLocked, return
```
Comment (seanyesmunt) [inline ui/redux/actions/settings.js:239]: Forgot this one?

File: ui/redux/actions/settings.js [hunk 4]
```diff
@@ -123,7 +147,19 @@ export function doSaveCustomWalletServers(servers) {
 }
 
 export function doSetClientSetting(key, value, pushPrefs) {
-  return dispatch => {
+  // not if sync locked
```
Comment (seanyesmunt) [inline ui/redux/actions/settings.js:150]: This comment isn't needed

File: ui/redux/actions/sync.js [hunk 5]
```diff
@@ -0,0 +1,343 @@
+import * as ACTIONS from 'constants/action_types';
```
Comment (seanyesmunt) [inline ui/redux/actions/sync.js:1]: Did you make any changes to this file?

**Review by seanyesmunt — CHANGES_REQUESTED** [review]
Minor comments

File: ui/redux/actions/sync.js (hunk 5 above)
Comment (jessopb) [reply ui/redux/actions/sync.js:1]: https://github.com/lbryio/lbry-redux/pull/355/commits/79f67889831a986b4f8a7b629c776b725f9614d2 here's a handy diff

Comment (jessopb) [reply ui/redux/actions/sync.js:1]: I brought the stuff from syncwrapper in.

Comment (jessopb) [reply ui/redux/actions/sync.js:1]: https://github.com/lbryio/lbry-desktop/pull/4882/commits/19e11742fb3ef92f4ab6a987052c10f6e6bac937

File: ui/redux/actions/settings.js (hunk 2 above)
Comment (jessopb) [reply ui/redux/actions/settings.js]: doAlertWaitingForSync in actions/app.js

**Comment by seanyesmunt** [comment]
Looks good. I'm going to wait to merge this until a patch is out for the desktop app

**Comment by kauffj** [comment]
This is mergeable

**Comment by tzarebczan** [comment]
Sync set is being called early on without an x auth token and fails

**Pull request merged**